### PR TITLE
style(python): use flake8 for all Python linting

### LINKED ISSUE #7051: RFC: switch Python docstring convention to Google-style
## overview

We currently specify our docstrings in the default, Sphinx/rST format:

```python
    @requires_version(2, 0)
    def load_labware(
            self,
            load_name: str,
            location: types.DeckLocation,
            label: str = None,
            namespace: str = None,
            version: int = None,
    ) -> Labware:
        """ Load a labware onto the deck given its name.

        For labware already defined by Opentrons, this is a convenient way
        # ...etc...

        :param load_name: A string to use for looking up a labware definition
        :param location: The slot into which to load the labware such as
                         1 or '1'
        :type location: int or str
        :param str label: An optional special name to give the labware. If
                          specified, this is the name the labware will appear
                          as in the run log and the calibration view in the
                          Opentrons app.
        :param str namespace: The namespace the labware definition belongs to.
            If unspecified, will search 'opentrons' then 'custom_beta'
        :param int version: The version of the labware definition. If
            unspecified, will use version 1.
        """
```

(If you go into some of our legacy code, you can also find NumPy-style comments... but let's ignore that).

Speaking for myself, I find these `:param`, `:type`, etc. directives a little difficult to write and even more difficult to read. **I would like to propose that we move over to [Google-style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)** to make these friendlier to write and read:

```python
    @requires_version(2, 0)
    def load_labware(
            self,
            load_name: str,
            location: types.DeckLocation,
            label: str = None,
            namespace: str = None,
            version: int = None,
    ) -> Labware:
        """Load a labware onto the deck given its name.

        For labware already defined by Opentrons, this is a convenient way
        # ...etc...

        Parameters:
            load_name: A string to use for looking up a labware definition.
            location: The slot into which to load the labware such as 1 or '1'.
            label: An optional special name to give the labware. If
                specified, this is the name the labware will appear
                as in the run log and the calibration view in the
                Opentrons app.
            namespace: The namespace the labware definition belongs to.
                If unspecified, will search 'opentrons' then 'custom_beta'.
            version: The version of the labware definition. If
                unspecified, will use version 1.

        Returns:
            The labware instance.
        """
```

## why?

My opinions on this matter:

- They're easier to read, with less cruft
- Separate sections for parameters, returns, examples are really nice, and this would enforce them
- Types of parameters are read from type annotations
- There's a larger ecosystem of doc generation tools that _aren't_ sphinx that are compatible with Google-style docstrings
    - Specifically, I'm thinking of https://github.com/pawamoy/mkdocstrings/
- pydocstyle can check for missing parameter descriptions in Google-style
    - Note: It can get tripped up pretty easily and fail to complain, but it's [actively being worked on](https://github.com/PyCQA/pydocstyle/milestone/6)

## how?

1. Switch the flake8-docstrings linting config from "pep257" to "google"
2. Ensure [sphinx.ext.napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) (already included in our Sphinx config) is properly configured for type annotations, etc.

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR follows up on the work done in #6945 and switches remaining Python projects to [flake8](https://flake8.pycqa.org/en/latest/), for consistency across all Opentrons' Python projects.

I also took the opportunity to set `flake8-annotations` to Google-style, which closes #7051.

(As we're moving into more Protocol Engine wire-up work, the linter discrepancy was getting in the way of ensuring work felt properly greenfield)

## Changelog

- Switch `robot-server` to flake8
- Switch `update-server` to flake8
- Switch `shared-data/python` to flake8
- Added `flake8-noqa` to yell about bad noqa comments
- Fix up obvious lints
    - Some docstrings
    - A lot of invalid `noqa` comments (or those made invalid by using flake8 instead of pylama)
    - Some stuff changed by `autopep8`

## Review requests

- [ ] Make sure I didn't touch any actual source code! (Only comments and docstrings)
- [ ] Changes look good and make sense

## Risk assessment

Very low, if both checkboxes above are true

### CODE REVIEW

File: api/src/opentrons/protocols/types.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def run(ctx):
 Please contact support@opentrons.com to retrieve the previous software version and be guided
 through the downgrade process.
 
-"""  # noqa E511
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocols/types.py:76]: Accidental number change `E511` -> `E501`?

File: api/src/opentrons/types.py [hunk 2]
```diff
@@ -3,14 +3,11 @@
 from math import sqrt, isclose
 from typing import Any, NamedTuple, TYPE_CHECKING, Union
 
-from opentrons.protocols.api_support.labware_like import LabwareLike
+from .protocols.api_support.labware_like import LabwareLike
 
 if TYPE_CHECKING:
-    from typing import (Optional,       # noqa: F401 Used for typechecking
-                        Tuple)
```
Comment (SyntaxColoring) [inline api/src/opentrons/types.py:10]: This is interesting—the linter wouldn't catch unused imports if they were guarded by an `if TYPE_CHECKING`, or something?

File: robot-server/.flake8 [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[flake8]
+
+# set line-length for future black support
+# https://github.com/psf/black/blob/master/docs/compatible_configs.md
+max-line-length = 88
```
Comment (SyntaxColoring) [inline robot-server/.flake8:5]: 🙌

File: robot-server/.flake8 [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    ANN102
+
+# configure flake8-docstrings
+# https://pypi.org/project/flake8-docstrings/
+docstring-convention = google
+
+noqa-require-code = true
+
+# TODO(mc, 2021-04-12): this config ignores type annotation and doc
+# string lints in these modules; remove entries as they are fixed
```
Comment (SyntaxColoring) [inline robot-server/.flake8:24]: Question: What "type annotation lints" are we ignoring? Mypy itself has its own configurables for erroring on the absence of type annotations. What else is there to check for?

**Review by sanni-t — APPROVED** [review]
- [x] Make sure I didn't touch any actual source code! (Only comments and docstrings)
 - [x] Changes look good and make sense
🙌

**Review by sfoster1 — APPROVED** [review]
Read through it and no code is touched, looks good to me!

**Review by SyntaxColoring — COMMENTED** [review]
Only read through about half the files, but what I saw looked good so far.

File: api/src/opentrons/protocols/types.py (hunk 1 above)
Comment (mcous) [reply api/src/opentrons/protocols/types.py:76]: Intentional change. E511 is not a thing, [E501 is line too long](https://www.flake8rules.com/rules/E501.html). I believe this was a case of an improperly formatted `noqa` (missing colon) turning into a blanket `noqa`

File: api/src/opentrons/types.py (hunk 2 above)
Comment (mcous) [reply api/src/opentrons/types.py:10]: It seems like `flake8` is smart enough to know typechecking imports are used. The two lines below triggered [NQA102 unused noqa](https://github.com/plinss/flake8-noqa/#error-codes) and these imports here from `typing` were actually completely unused

File: robot-server/.flake8 (hunk 4 above)
Comment (mcous) [reply robot-server/.flake8:24]: See [flake8-annotations](https://github.com/sco1/flake8-annotations). TL;DR: `mypy` checks your type annotations, `flake8` checks _that_ you added type annotations to your code in the first place

Comment (SyntaxColoring) [reply robot-server/.flake8:24]: > `flake8` checks that you added type annotations to your code in the first place

[So can mypy though](https://mypy.readthedocs.io/en/stable/config_file.html?highlight=configuration#confval-disallow_untyped_defs), right? Not a problem for this PR, but I guess I don't understand what flake8-annotations adds.

**Review by amitlissack — APPROVED** [review]

**Pull request merged**